controlplane/internal/controller/agentclusterinstall_controller.go

```diff
@@ -58,78 +53,70 @@ func (r *AgentClusterInstallReconciler) Reconcile(ctx context.Context, req ctrl.
 	log := ctrl.LoggerFrom(ctx)
 
 	defer func() {
-		log.Info("Agent Cluster Install Reconcile ended")
+		log.V(logutil.TraceLevel).Info("Agent Cluster Install Reconcile ended")
 	}()
 
-	log.Info("Agent Cluster Install Reconcile started")
+	log.V(logutil.TraceLevel).Info("Agent Cluster Install Reconcile started")
 	aci := &hiveext.AgentClusterInstall{}
 	if err := r.Client.Get(ctx, req.NamespacedName, aci); err != nil {
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
 		return ctrl.Result{}, err
 	}
-	log.Info("Reconciling AgentClusterInstall", "name", aci.Name, "namespace", aci.Namespace)
+	log.WithValues("AgentClusterInstall Name", aci.Name, "AgentClusterInstall Namespace", aci.Namespace)
 
-	cd, err := r.getClusterDeployment(ctx, aci)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			return ctrl.Result{}, nil
-		}
+	acp := controlplanev1alpha1.AgentControlPlane{}
+	if err := util.GetTypedOwner(ctx, r.Client, aci, &acp); err != nil {
 		return ctrl.Result{}, err
 	}
-	acpList, err := r.getAgentControlPlaneList(ctx, cd, aci)
-	if err != nil {
+	log.WithValues("AgentControlPlane Name", acp.Name, "AgentControlPlane Namespace", acp.Namespace)
+
+	if err := r.reconcileInitialized(ctx, aci, &acp); err != nil {
 		return ctrl.Result{}, err
 	}
-	if acpList == nil || len(acpList.Items) == 0 {
-		log.Info("agentcontrolplane not found", "num", len(acpList.Items))
-		return ctrl.Result{}, nil
-	}
-	log.Info("found agentcontrolplane", "num", len(acpList.Items))
 
-	for _, acp := range acpList.Items {
-		// Check if AgentClusterInstall has moved to day 2 aka control plane is installed
-		if isInstalled(aci) && hasKubeconfigRef(aci) {
-			acp.Status.Initialized = true
+	// Check if AgentClusterInstall has moved to day 2 aka control plane is installed
+	if isInstalled(aci) {
+		acp.Status.Ready = true
+		if err := r.Client.Status().Update(ctx, &acp); err != nil {
+			log.Error(err, "failed setting AgentControlPlane to ready")
+			return ctrl.Result{}, err
+		}
+	}
+	return ctrl.Result{}, nil
+}
 
-			log.Info("Agent cluster install for control plane nodes has finished. Attaching kubeconfig secret to agent control plane",
-				"agent cluster install name", aci.Name, "agent cluster install namespace", aci.Namespace, "agent control plane name", acp.Name)
+func (r *AgentClusterInstallReconciler) reconcileInitialized(ctx context.Context, aci *hiveext.AgentClusterInstall, acp *controlplanev1alpha1.AgentControlPlane) error {
```

Comment:
Not sure about this name, why `reconcileInitialized`? maybe just `reconcile` ?

Reply:
Just because this only checks for `initialized` status but sure I can change it to just `reconcile`